### PR TITLE
Adie script for AIO on AKS-EE

### PR DESCRIPTION
**Pull request by angop95**
-  Add a dedicated aide script for AIO on AKS-EE with default resources suitable for AIO on AKSEE
- Ensure all error statements are displayed in red foreground color

### CODE REVIEW

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+$starttime = Get-Date
+$starttimeString = $($starttime.ToString("yyMMdd-HHmm"))
+$transcriptFile = "$installDir\aksedgedlog-$starttimeString.txt"
+
+Start-Transcript -Path $transcriptFile
+
+Set-ExecutionPolicy Bypass -Scope Process -Force
+# Download the AksEdgeDeploy modules from Azure/AksEdge
+$fork ="angop95"
```
Comment (fcabrera23) [inline tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:112]: We shoudl probably change this to official repo and not  your fork

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    "SchemaVersion": "1.1",
+    "Version": "1.0",
+    "AksEdgeProduct": "$productName",
+    "AksEdgeProductUrl": "",
+    "Azure": {
+        "SubscriptionName": "",
+        "SubscriptionId": "$SubscriptionId",
+        "TenantId": "$TenantId",
+        "ResourceGroupName": "angop-test-aksedge-rg",
+        "ServicePrincipalName": "angop-test-aksedge-sp",
```
Comment (fcabrera23) [inline tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:56]: Update to use variable

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        "Location": "$Location",
+        "CustomLocationOID":"",
+        "Auth":{
```
Comment (fcabrera23) [inline tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:59]: @parameshbabu i think we wanted to remove the requirement of service Principal - Could you please further advice on how to proceed?

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    [String] $TenantId,
+    [ValidateNotNullOrEmpty()]
+    [String] $Location,
+    [Switch] $UseK8s,
+    [string] $Tag
+)
+#Requires -RunAsAdministrator
+New-Variable -Name gAksEdgeQuickStartVersion -Value "1.0.231016.1400" -Option Constant -ErrorAction SilentlyContinue
+
+New-Variable -Option Constant -ErrorAction SilentlyContinue -Name arcLocations -Value @(
```
Comment (fcabrera23) [inline tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:17]: Can we check with AIO team to confirm all these locations will be supported with AIO?

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    "AksEdgeProduct": "$productName",
+    "AksEdgeProductUrl": "",
+    "Azure": {
+        "SubscriptionName": "",
+        "SubscriptionId": "$SubscriptionId",
+        "TenantId": "$TenantId",
+        "ResourceGroupName": "angop-test-aksedge-rg",
+        "ServicePrincipalName": "angop-test-aksedge-sp",
+        "Location": "$Location",
+        "CustomLocationOID":"",
```
Comment (fcabrera23) [inline tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:58]: CustomLocationOID is now supported - Check with Sean

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+catch {
+    Write-Host "Error: Firewall rule addition for AIO MQTT broker failed" -ForegroundColor Red
+    Stop-Transcript | Out-Null
+    Pop-Location
+    exit -1 
+}
+
+Write-Host "Configuring port proxy for AIO"
+try {
+    netsh interface portproxy add v4tov4 listenport=1883 listenaddress=0.0.0.0 connectport=1883 connectaddress=192.168.0.4 | Out-Null
```
Comment (fcabrera23) [inline tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:226]: 192.168.0.4 -> This is the address if the 192.168.0.X family is available,. but could change - Probably we need to check and use the appropriate subnet

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 4 above)
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:17]: Sure, let me check on the location support for AIO.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 6 above)
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:226]: Let me check how we pick subnet in AKS-EE and see if we can extend the same logic here.

Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: Using the `Get-AksEdgeDeploymentInfo` command to get the service ip start and use it.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 5 above)
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:56]: Spoke to Sean, we don't have the `CustomLocationOID` changes are merged yet.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 2 above)
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: Replaced it with the default values from the existing quickstart script.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 1 above)
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: updated

**Review by parameshbabu — CHANGES_REQUESTED** [review]

**Review by parameshbabu — APPROVED** [review]

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+} else {
+    Write-Host -Message "Deployment failed" -Category OperationStopped -ForegroundColor Red
+    Stop-Transcript | Out-Null
+    Pop-Location
+    exit -1
+}
+
+Write-Host "Step 4: Connect to Arc"
+if ($skipAzureArc) {
+    Write-Host ">> skipping step 4" -ForegroundColor Yellow
```
[the remark this answers is not in the document]
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: Updated the script to always connect to ARC and without the `skipAzureArc` flag.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+            Pop-Location
+            exit -1
+        }
+    } else { Write-Host "Error: Arc Initialization failed. Skipping Arc Connection" -ForegroundColor Red }
+}
+
+Write-Host "Step 5: Prep for AIO workload deployment"
+Write-Host "Deploy local path provisioner"
+try {
+    & 'C:\Program Files\AksEdge\kubectl\kubectl.exe' apply -f 'https://raw.githubusercontent.com/Azure/AKS-Edge/main/samples/storage/local-path-provisioner/local-path-storage.yaml'
```
[the remark this answers is not in the document]
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: Removed the hardcoded path and using the `kubectl` binary directly from the path environment.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+            Stop-Transcript | Out-Null
+            Pop-Location
+            exit -1
+        }
+    } else { Write-Host "Error: Arc Initialization failed. Skipping Arc Connection" -ForegroundColor Red }
+}
+
+Write-Host "Step 5: Prep for AIO workload deployment"
+Write-Host "Deploy local path provisioner"
+try {
```
[the remark this answers is not in the document]
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:196]: Added additional check to ensure we are not duplicating the Firewall, iptable rules. 
The `kubectl apply` command is idempotent by design so living that as is.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 8 above)
[the remark this answers is not in the document]
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: kubectl apply does allow to specify the URI directly without requiring a local copy. Any reason why we should use a local copy?

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+    [String] $SubscriptionId,
+    [ValidateNotNullOrEmpty()]
+    [String] $TenantId,
+    [ValidateNotNullOrEmpty()]
+    [String] $Location,
+    [Switch] $UseK8s,
+    [string] $Tag
+)
+#Requires -RunAsAdministrator
+New-Variable -Name gAksEdgeQuickStartVersion -Value "1.0.231016.1400" -Option Constant -ErrorAction SilentlyContinue
```
[the remark this answers is not in the document]
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: Updated the name

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 8 above)
[the remark this answers is not in the document]
Comment (parameshbabu) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: avoid another file download from internet while the same copy is locally available.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 3 above)
Comment (parameshbabu) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1:56]: Lets take this offline and explore. Not gate this commit on this.

File: tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1 (hunk 8 above)
[the remark this answers is not in the document]
Comment (angop95) [reply tools/scripts/AksEdgeQuickStart/AksEdgeQuickStartForAio.ps1]: updated to use local copy of the sample.

**Review by fcabrera23 — APPROVED** [review]
LGTM

**Pull request merged**